src/test/java/com/google/api/codegen/testdata/java_sample_library.baseline

```diff
@@ -50,3 +52,131 @@ jar {
+          authentication(userName: ossrhUsername, password: ossrhPassword)
+        }
+
+        pom.project {
+          name "com.google.api.grpc:google-cloud-library-v1"
+          description project.description
+          url 'https://github.com/googleapis/googleapis'
+          scm {
+            url 'https://github.com/googleapis/googleapis'
+            connection 'scm:git:https://github.com/googleapis/googleapis'
+          }
+
+          licenses {
+            license {
+              name 'Apache-2.0'
+              url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+            }
+          }
+
+          developers {
+            developer {
+              id 'Google, Inc.'
+              name 'Google, Inc.'
+              email 'googleapis-packages@google.com'
+              url 'https://github.com/googleapis/googleapis'
+              organization 'Google, Inc.'
+              organizationUrl 'https://www.google.com'
+            }
+          }
+        }
+      }
+    }
+  }
+} else {
+  task failUploadArchives << {
+    throw new GradleException("uploadArchives needs ossrhUsername and ossrhPassword "
+      + "to be set. They can be set inside your ~/.gradle/gradle.properties file.")
+  }
+  tasks.uploadArchives.dependsOn(failUploadArchives)
+}
+
+task checkOutGhPages << {
+  if (!new File(javaDocRoot).exists()) {
+    exec {
+      workingDir projectDir
+      commandLine 'git', 'clone', '--branch', 'gh-pages',
+          '--single-branch', 'git@github.com:googleapis/googleapis.git', 'tmp_gh-pages'
+    }
+  }
+}
+
+task copyFilesToGhPages {
+  dependsOn 'checkOutGhPages'
+  dependsOn 'javadoc'
+  doLast {
+    def newSiteDirPath = javaDocRoot + "/java/${packageName}/${project.version}/apidocs/"
+    new File(newSiteDirPath).mkdirs()
+    copy {
+      from 'build/docs/javadoc'
+      into newSiteDirPath
+    }
+  }
+}
+
+// Regenerates the gh-pages branch under tmp_gh-pages
+task updateDocs {
+  dependsOn 'copyFilesToGhPages'
+  doLast {
+    def outputContent = new File(projectDir.getAbsolutePath() + '/templates/apidocs_index.html.template').text
+    outputContent = outputContent.replace('{{siteVersion}}', project.version)
+    outputContent = outputContent.replace('{{packageName}}', packageName)
+    new File(javaDocRoot + "/java/${packageName}/apidocs").mkdirs()
+    new File(javaDocRoot + "/java/${packageName}/apidocs/index.html").write(outputContent)
+    exec {
+      workingDir javaDocRoot
+      commandLine 'git', 'add', '.'
+    }
+    exec {
+      workingDir javaDocRoot
+      commandLine 'git', 'commit', '-m', "Regenerating docs for ${packageName} ${project.version}"
+    }
+    println 'New docs have been generated under tmp_gh-pages and have been committed;'
+    println 'Next they just need to be pushed with "git push".'
+  }
+}
+
+// Regenerates and push the gh-pages branch under tmp_gh-pages
+task updateAndPushDocs {
```

Comment:
This is an idea for later - maybe we should make our own plugin so we don't have to repeat this tasks in every single build.gradle we own.